packages/fx-core/src/core/index.ts

```diff
@@ -277,6 +281,40 @@ export class FxCore implements Core {
     return ok(projectPath);
   }
 
+  async archive(projectPath: string): Promise<void> {
+    const archiveFolderPath = path.join(projectPath, ArchiveFolderName);
+    await fs.ensureDir(archiveFolderPath);
+
+    const fileNames = await fs.readdir(projectPath);
+    const archiveLog = async (projectPath: string, message: string): Promise<void> => {
+      await fs.appendFile(
+        path.join(projectPath, ArchiveLogFileName),
+        `[${new Date().toISOString()}] ${message}\n`
+      );
+    };
+
+    await archiveLog(projectPath, `Start to move files into '${ArchiveFolderName}' folder.`);
+    for (const fileName of fileNames) {
+      if (fileName === ArchiveFolderName || fileName === ArchiveLogFileName) {
+        continue;
```

Comment:
Exclude `node_modules`?

Reply:
We move `node_modules` into `.archive` folder, but we do not copy `node_modules` into the new project

Reply:
Not sure if `node_modules` should be moved to `.archive` folder, but I think it is OK since `move` operation is very fast.